src/replication/ng/registry/manager.go

```diff
@@ -202,15 +202,16 @@ func (m *DefaultManager) HealthCheck() error {
 	return nil
 }
 
-func healthStatus(r *model.Registry) (HealthStatus, error) {
+// CheckHealthStatus checks status of a given registry
+func CheckHealthStatus(r *model.Registry) (HealthStatus, error) {
 	// TODO(ChenDe): Support other credential type like OAuth, for the moment, only basic auth is supported.
 	if r.Credential.Type != model.CredentialTypeBasic {
 		return Unknown, fmt.Errorf("unknown credential type '%s', only '%s' supported yet", r.Credential.Type, model.CredentialTypeBasic)
 	}
 
 	// TODO(ChenDe): Support health check for other kinds of registry
 	if r.Type != model.RegistryTypeHarbor {
-		return Unknown, fmt.Errorf("unknown registry type '%s'", model.RegistryTypeHarbor)
+		return Unknown, fmt.Errorf("unknown registry type '%s'", r.Type)
```

Comment:
As we discussed offline, we need to put the `heath check` or `ping` method to the `Adapter` interface and call the method when do ping.

Reply:
How about do it in another PR.

Reply:
Sure

Reply:
I will add Ping to Adapter in this PR.